### DISCUSSION TITLE
Should config options only be considered for an initial start/creation or for every successive run

### DISCUSSION THREAD

**Discussion by gbraad** [crc-org/crc#2317] [Q&A]
When CRC is used to start a cluster with

```
$ crc start --disk-size 40G
$ crc stop
$ crc config set disk-size 60G
$ crc start
```

Should this resize the disk?
Note: this is to refine the user story for #2243 and #2261

**Comment by jsliacan** [comment]
Your scenario above *does* resize the disk. Same goes for memory and number of cpus. Also, other config options are applied at that point too, like skip-checks. The problem shows up in some cases when a setting is unset after `start` and `stop`. 

Default config settings are config settings, only implicit. If the user is going through a process when config settings are applied (e.g. start after stop), the settings that are being applied should be taken with priority from flags, config file (and env vars?).  If this is not the case, it's unexpected. 

Disk size config option is special in that it's directional, which makes the scenario above awkward if you go down in size - not possible. But the user is informed if the setting is explicit in the config file (of if a flag is passed). If, however, `disk-size` is simply unset, the implicit disk-size of 31G is not respected and user is not informed about this. It's just a simple oversight, works in all other cases.

**Reply by gbraad** [reply]
Why I bring this up is the following. If you say this 'does' resize, it am not sure what to expect from:
```
$ crc config unset disk-size
$ crc start
```
as we can not resize to a smaller size, but the 'default' value would have to imply this... we silently ignore this? or give an error this is not possible? This is more explicit if done as:

```
$ crc config set disk-size 35G
$ crc start
```

because in that case you shouldn't ignore it, but error out? throw an info?

Note: I do not think we should treat 'default' values differently from actually set values, but I do not think we need to 'store' them in a file (as they need to be enforced without someone being able to modify that upper/lower bound).

**Reply by jsliacan** [reply]
> Why I bring this up is the following. If you say this 'does' resize, it am not sure what to expect from:
> $ crc config unset disk-size
> $ crc start

That's exactly the issue that started this discussion: #2315. I think that in this case, like it already is in other cases, the user should be informed that the requested resize is not possible and `crc start` should error out. 

> I do not think we need to 'store' them in a file

We don't need to, but if we do, there's a whole host of nice consequences. We could simplify the entire config setting protocol for the user. And of course, simple => fewer user errors  && fewer bugs on our side.

**Comment by jsliacan** [comment]
Maybe we could remove all flags and write all default settings to config file (as if the user set them at the beginning of their interaction with CRC). So `crc.json` would be the only source of configuration settings, and it would be complete -- what's not in it is not applied (no implicit defaults). This would significantly simplify this whole thing.
